compiler/rustc_lint/src/unnecessary_block.rs

```diff
@@ -0,0 +1,45 @@
+use crate::lints::UnnecessaryBlock as UnnecessaryBlockErr;
+use crate::{LateContext, LateLintPass, LintContext};
+use rustc_hir::{BlockCheckMode, ExprKind, Stmt, StmtKind};
+
+declare_lint! {
+    /// The `unnecessary_block` lint checks for unnecessary block.
```

Comment:
It might be beneficial to explain which blocks are considered unnecessary:
```suggestion
    /// The `unnecessary_block` lint checks for empty blocks in statement position.
```